projects/plugins/automattic-for-agencies-client/src/js/components/disconnected-card/index.jsx

```diff
@@ -0,0 +1,61 @@
+import { CONNECTION_STORE_ID, ConnectButton } from '@automattic/jetpack-connection';
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import classNames from 'classnames';
+import React, { useMemo } from 'react';
+import BrandedCard from '../branded-card';
+import CloseCircleIcon from '../close-circle-icon';
+import styles from './styles.module.scss';
+
+/**
+ * Disconnected Card component.
+ *
+ * @returns {React.Component} The `ConnectionCard` component.
+ */
+export default function DisconnectedCard() {
+	const { apiNonce, apiRoot, registrationNonce } = window.automatticForAgenciesClientInitialState;
+
+	const connectionErrors = useSelect(
+		select => select( CONNECTION_STORE_ID ).getConnectionErrors(),
+		[]
+	);
+
+	// Use the highest-level error message.
+	const errorMessage = useMemo( () => {
+		return Object.values( connectionErrors )?.shift()?.shift()?.error_message;
+	}, [ connectionErrors ] );
```

Comment:
Out of curiosity, the way we are using `shift` to get the highest-level error message relies on the assumption that the array contains two error objects at max. Is that a correct interpretation of this code?

Reply:
To my understanding, the `connectionErrors` is an object of key-value pairs, where the values are actually arrays. I'm assuming this is to support multiple errors under the same key/category. For example:
```
{
  error_key_a: [
    { ...error },
    { ...error },
    { ...error },
  ],
  error_key_b: [
    { ...error },
  ],
}
```
So we end up working with an array of arrays, hence the two `shift()`s here.